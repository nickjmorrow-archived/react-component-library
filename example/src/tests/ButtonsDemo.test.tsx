import * as React from "react";
import { Buttons } from "../components/demos/Buttons";
import { createRenderer } from "react-test-renderer/shallow";

const renderer = createRenderer();
it("renders correctly", () => {
  const tree = renderer.render(<Buttons />);
  expect(tree).toMatchSnapshot();
});
