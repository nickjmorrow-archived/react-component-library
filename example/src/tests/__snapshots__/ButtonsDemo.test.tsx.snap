// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Unknown>
  <Unknown
    customStyle={
      Object {
        "backgroundColor": "hsl(200, 13%, 94%)",
        "display": "none",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "paddingBottom": "48px",
        }
      }
    >
      <Typography
        sizeVariant={4}
      >
        Change Paper Color
      </Typography>
      <Select
        currentOption={
          Object {
            "label": "light",
            "value": "hsl(200, 13%, 94%)",
          }
        }
        onChange={[Function]}
        options={
          Array [
            Object {
              "label": "light",
              "value": "hsl(200, 13%, 94%)",
            },
            Object {
              "label": "medium",
              "value": "hsl(200, 13%, 55%)",
            },
            Object {
              "label": "dark",
              "value": "hsl(200, 13%, 45%)",
            },
          ]
        }
      />
    </div>
  </Unknown>
  <div>
    <Typography
      sizeVariant={6}
      style={
        Object {
          "marginBottom": "16px",
        }
      }
      weightVariant={2}
    >
      Style Variants
    </Typography>
  </div>
  <Typography
    colorVariant="secondaryDark"
    sizeVariant={5}
    style={
      Object {
        "marginBottom": "12px",
      }
    }
  >
    Primary
  </Typography>
  <Unknown
    customStyle={
      Object {
        "backgroundColor": "hsl(200, 13%, 94%)",
      }
    }
  >
    <Unknown
      colorVariant="core"
    >
      Core
    </Unknown>
    <Unknown
      colorVariant="accent"
    >
      Accent
    </Unknown>
    <Unknown
      colorVariant="success"
    >
      Success
    </Unknown>
    <Unknown
      colorVariant="warning"
    >
      Warning
    </Unknown>
    <Unknown
      colorVariant="danger"
    >
      Danger
    </Unknown>
  </Unknown>
  <Typography
    colorVariant="secondaryDark"
    sizeVariant={5}
    style={
      Object {
        "marginBottom": "12px",
      }
    }
  >
    Secondary
  </Typography>
  <Unknown
    customStyle={
      Object {
        "backgroundColor": "hsl(200, 13%, 94%)",
      }
    }
  >
    <Unknown
      colorVariant="core"
      styleVariant="secondary"
      textColorVariant="core"
      useMargin={true}
    >
      Core
    </Unknown>
    <Unknown
      colorVariant="accent"
      styleVariant="secondary"
      textColorVariant="accent"
      useMargin={true}
    >
      Accent
    </Unknown>
    <Unknown
      colorVariant="success"
      styleVariant="secondary"
      textColorVariant="success"
      useMargin={true}
    >
      Success
    </Unknown>
    <Unknown
      colorVariant="warning"
      styleVariant="secondary"
      textColorVariant="warning"
      useMargin={true}
    >
      Warning
    </Unknown>
    <Unknown
      colorVariant="danger"
      styleVariant="secondary"
      textColorVariant="danger"
      useMargin={true}
    >
      Danger
    </Unknown>
  </Unknown>
  <Typography
    colorVariant="secondaryDark"
    sizeVariant={5}
    style={
      Object {
        "marginBottom": "12px",
      }
    }
  >
    Tertiary
  </Typography>
  <Unknown
    customStyle={
      Object {
        "backgroundColor": "hsl(200, 13%, 94%)",
      }
    }
  >
    <Unknown
      colorVariant="core"
      styleVariant="tertiary"
      textColorVariant="core"
    >
      Core
    </Unknown>
    <Unknown
      colorVariant="accent"
      styleVariant="tertiary"
      textColorVariant="accent"
    >
      Accent
    </Unknown>
    <Unknown
      colorVariant="success"
      styleVariant="tertiary"
      textColorVariant="success"
    >
      Success
    </Unknown>
    <Unknown
      colorVariant="warning"
      styleVariant="tertiary"
      textColorVariant="warning"
    >
      Warning
    </Unknown>
    <Unknown
      colorVariant="danger"
      styleVariant="tertiary"
      textColorVariant="danger"
    >
      Danger
    </Unknown>
  </Unknown>
  <div>
    <Typography
      sizeVariant={6}
      style={
        Object {
          "marginBottom": "16px",
        }
      }
      weightVariant={2}
    >
      Other Props
    </Typography>
  </div>
  <Typography
    colorVariant="secondaryDark"
    sizeVariant={5}
    style={
      Object {
        "marginBottom": "12px",
      }
    }
  >
    Loading
  </Typography>
  <Unknown
    customStyle={
      Object {
        "backgroundColor": "hsl(200, 13%, 94%)",
      }
    }
  >
    <Unknown
      isLoading={false}
      onClick={[Function]}
      textColorVariant="primaryLight"
    >
      I'm loading
    </Unknown>
    <Unknown
      isLoading={false}
      onClick={[Function]}
      styleVariant="secondary"
      textColorVariant="core"
    >
      I'm loading
    </Unknown>
  </Unknown>
  <Typography
    colorVariant="secondaryDark"
    sizeVariant={5}
    style={
      Object {
        "marginBottom": "12px",
      }
    }
  >
    Link Buttons
  </Typography>
  <Unknown
    customStyle={
      Object {
        "backgroundColor": "hsl(200, 13%, 94%)",
      }
    }
  >
    <Unknown
      colorVariant="accent"
      link="google.com"
    >
      Google
    </Unknown>
    <Unknown
      link="facebook.com"
    >
      Facebook
    </Unknown>
    <Unknown
      colorVariant="success"
      link="instagram.com"
    >
      Instagram
    </Unknown>
  </Unknown>
</Unknown>
`;
